documentation/docs/backward-compatibility.md

```diff
@@ -3,43 +3,26 @@ layout: doc
 title: Backward compatibility&#58;
 ---
 
-Living projects need changes and sometimes these are breaking ones. Breaking changes are generally available in major versions.
+## 1) Versioning & backward compatibility
 
-`pac4j` is a whole project in itself and all *pac4j* implementations are also projects in themselves which depend on `pac4j` (in fact, they have a dependency on the `pac4j-core` library). And this generates a huge number of constraints and work to keep a full backward compatibility.
+Since version 2, the `pac4j` project has adopted the [semver](http://semver.org/) versioning, to make things clearer.
 
-That's why the *pac4j* project has a specific definition when it comes to backward compatibility. We want to go fast and we make compromises to achieve this goal.
+Given a version X.Y.Z:
 
+- when X changes, it's a major version with breaking changes (example: 2.x, 3.x, and so on)
+- when Y changes, it's a minor version with no compilation / runtime breaking changes (for example, you can upgrade from 2.5.2 to 2.7.0 without any worry)
+- when Z changes, it's a bug fix(es) version with no compilation / runtime breaking change (no upgrade issue either).
 
-## 1) Versioning
+*pac4j* implementations will adopt the semver versioning when upgrading to `pac4j` version 2.
```

Comment:
yes, so it also means that if pac4j has a breaking change (bump the major number), the implementation that uses it should too if it upgrades pac4j, since it kind of exposes pac4j APIs, right?